app-modules/contact/database/migrations/2024_06_09_122855_data_seed_permissions_for_organization_module.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Aiding App™ is licensed under the Elastic License 2.0. For more details,
+    see <https://github.com/canyongbs/aidingapp/blob/main/LICENSE.>
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Migrations\Migration;
+use Database\Migrations\Concerns\CanModifyPermissions;
+
+return new class () extends Migration {
+    use CanModifyPermissions;
+
+    private array $permissions = [
+        'organization.view-any' => 'Organization',
+        'organization.create' => 'Organization',
+        'organization.*.view' => 'Organization',
+        'organization.*.update' => 'Organization',
+        'organization.*.delete' => 'Organization',
+        'organization.*.restore' => 'Organization',
+        'organization.*.force-delete' => 'Organization',
+    ];
+
+    private array $guards = [
+        'web',
+        'api',
+    ];
+
+    public function up(): void
+    {
+        collect($this->guards)
+            ->each(function (string $guard) {
+                $permissions = Arr::except($this->permissions, keys: DB::table('permissions')
+                    ->where('guard_name', $guard)
+                    ->pluck('name')
+                    ->all());
+
```

Comment:
@danharrin though I understand what you mean, I feel it is safer to leave it as is. Even if it would never do so, this will just add additional protection to ensure there is no possibility of a unique constraint violation being thrown.

Reply:
I would agree generally if this was app code, but this adds another query to the migration process to fetch every permission in the DB and pluck their names

Reply:
As discussed outside this PR. Let's leave it how it is for now. We will standardize how permission seeding is written and then go back and make them all the same, later.

Reply:
Ah, whoops I made this comment without reloading the page so I didn't see this conversation.